server/sonar-ce/src/main/java/org/sonar/ce/taskprocessor/CeWorkerImpl.java

```diff
@@ -135,15 +138,27 @@ private void executeTask(CeTask task) {
         status = CeActivityDto.Status.FAILED;
       }
     } catch (MessageException e) {
+      // error
       error = e;
     } catch (Throwable e) {
+      // error
       LOG.error(format("Failed to execute task %s", task.getUuid()), e);
       error = e;
     } finally {
       finalizeTask(task, ceProfiler, status, taskResult, error);
     }
   }
 
+  private void callListeners(Consumer<ExecutionListener> call) {
+    listeners.forEach(listener -> {
+      try {
+        call.accept(listener);
+      } catch (Throwable t) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1181)